src/util-byte.c

```diff
@@ -360,6 +360,107 @@ int StringParseUint8(uint8_t *res, int base, uint16_t len, const char *str)
 
     return ret;
 }
+
+int StringParseUint64wValidation(uint64_t *res, int base, uint16_t len, const char *str,
+                                 uint64_t min, uint64_t max)
+{
+    int ret;
+
+    ret = ByteExtractString(res, base, len, str, true);
+    if (ret <= 0) {
+        return ret;
+    }
+
+    if ((uint64_t)ret < min || (uint64_t)ret > max) {
```

Comment:
`ret` is int, so it can't hold the full value range of `uint64_t`

Reply:
Should that be `*res` instead?

Reply:
Yes, my bad. It should be *res everywhere.